tests/unit/config_test.py

```diff
@@ -39,6 +39,42 @@ def test_config_validation(self):
         config.make_service_dict('foo', {'ports': ['8000']})
 
 
+class MergeTest(unittest.TestCase):
+    def test_merge_volumes(self):
+        service_dict = config.merge_service_dicts({}, {})
+        self.assertNotIn('volumes', service_dict)
+
+        service_dict = config.merge_service_dicts(
+            {'volumes': ['/foo:/code', '/data']},
+            {},
+        )
+        self.assertEqual(set(service_dict['volumes']), set(['/foo:/code', '/data']))
+
+        service_dict = config.merge_service_dicts(
+            {},
+            {'volumes': ['/bar:/code']},
+        )
+        self.assertEqual(set(service_dict['volumes']), set(['/bar:/code']))
+
+        service_dict = config.merge_service_dicts(
+            {'volumes': ['/foo:/code', '/data']},
+            {'volumes': ['/bar:/code']},
+        )
+        self.assertEqual(set(service_dict['volumes']), set(['/bar:/code', '/data']))
+
+        service_dict = config.merge_service_dicts(
+            {'volumes': ['/foo:/code', '/data']},
+            {'volumes': ['/bar:/code', '/quux:/data']},
+        )
+        self.assertEqual(set(service_dict['volumes']), set(['/bar:/code', '/quux:/data']))
+
+        service_dict = config.merge_service_dicts(
+            {'volumes': ['/foo:/code', '/quux:/data']},
+            {'volumes': ['/bar:/code', '/data']},
+        )
+        self.assertEqual(set(service_dict['volumes']), set(['/bar:/code', '/data']))
+
```

Comment:
Normally this would be 6 separate tests, so the test names can describe the case that is being tested.  It's not completely clear to which the difference between each of these without inspecting each in detail.